### PR TITLE
feat(dashboard): Clerk based settings page

### PR DESCRIPTION
**Pull request by scopsy**
### What changed? Why was the change needed?
- Added a new settings page with tabs
- Filled clerk components for profile and organization pages

### Screenshots
CleanShot 2024-12-03 at 16 13 40@2x

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/primitives/tabs.tsx [hunk 1]
```diff
@@ -4,23 +4,29 @@ import * as TabsPrimitive from '@radix-ui/react-tabs';
...
     variant: {
-      default: 'h-9 justify-center rounded-[10px] bg-neutral-alpha-100 p-1 text-muted-foreground',
+      default: 'h-9 rounded-[10px] bg-neutral-alpha-100 p-1 text-muted-foreground',
       regular: 'border-neutral-alpha-200 w-full justify-start gap-6 border-b border-t px-3.5',
     },
+    align: {
+      center: 'justify-center',
+      start: 'justify-start',
+      end: 'justify-end',
+    },
```
Comment (scopsy) [inline apps/dashboard/src/components/primitives/tabs.tsx:17]: Added control on tabs alignment

**Comment by SokratisVidros** [comment]
@scopsy please update the <UserButton/> as well so that its settings option points to our settings page, not the default Clerk modal.

Screenshot 2024-12-04 at 16 26 08

File: apps/dashboard/src/pages/settings.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        break;
+    }
+  };
+
+  return (
+    <DashboardLayout headerStartItems={<h1 className="text-foreground-950">Settings</h1>}>
+      <Tabs value={currentTab} onValueChange={handleTabChange} className="w-full">
+        <TabsList
+          align="center"
+          className="border-border/20 relative mt-4 flex w-full items-end justify-start space-x-2 rounded-none border-b bg-transparent px-1.5 pb-0"
```
Comment (SokratisVidros) [inline apps/dashboard/src/pages/settings.tsx]: How about placing it to the right? I feel it looks a bit better.

Screenshot 2024-12-04 at 16 28 26

File: apps/dashboard/src/pages/settings.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import { Card } from '@/components/primitives/card';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/primitives/tabs';
+import { OrganizationProfile, UserProfile } from '@clerk/clerk-react';
+import { DashboardLayout } from '../components/dashboard-layout';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { ROUTES } from '@/utils/routes';
+
+export const clerkComponentAppearance = {
+  elements: {
```
Comment (SokratisVidros) [inline apps/dashboard/src/pages/settings.tsx:22]: Please update the primary font color (black) to match the Dashboard primary font color `rgba(82, 88, 102, 0.95)`

File: apps/dashboard/src/pages/settings.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            Team
+          </TabsTrigger>
+        </TabsList>
+
+        <div className="mx-auto mt-1 max-w-[700px] px-1.5">
+          <TabsContent value="account" className="rounded-lg">
+            <Card className="mx-auto border-none shadow-none">
+              <UserProfile appearance={clerkComponentAppearance}>
+                <UserProfile.Page label="account" />
+                <UserProfile.Page label="security" />
```
Comment (SokratisVidros) [inline apps/dashboard/src/pages/settings.tsx]: For posterity: This issue has been reported to Clerk.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]
@scopsy Can you also verified that the connect oauth functionality works? It entails a redirect to OAuth and then back to this app.

File: apps/dashboard/src/pages/settings.tsx (hunk 2 above)
Comment (scopsy) [reply apps/dashboard/src/pages/settings.tsx]: Evaluated multiple version with Naveen, decided to go with central as overall we thought it looked best until we review this again.

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**